test/gtest/cpu/algorithms.cpp

```diff
@@ -36,6 +36,15 @@ void check_copy(std::size_t n, std::size_t b, std::size_t e) {
   if (comm_rank == 0) {
     std::copy(dv_in.begin() + b, dv_in.begin() + e, dv3.begin() + b);
   }
+  if (comm_rank == 0) {
+    lib::copy(0, v_in.begin() + b, v_in.begin() + e, dv8.begin() + b);
+  } else {
+    lib::copy(0, nullptr, nullptr, dv8.begin() + b);
+  }
+
+  lib::copy(0, comm_rank == 0 ? &*(v_in.begin() + b) : nullptr,
+            comm_rank == 0 ? &*(v_in.begin() + e) : nullptr, dv9.begin() + b);
```

Comment:
Unnecessary obfuscation:
```cpp
 lib::copy(0, comm_rank == 0 ? &*(v_in.begin() + b) : nullptr,
            comm_rank == 0 ? &*(v_in.begin() + e) : nullptr, dv9.begin() + b);
```
=>
```cpp
if (comm_rank == 0) {
    lib::copy(0, &*(v_in.begin() + b), &*(v_in.begin() + e), dv9.begin() + b);
} else {
    lib::copy(0, nullptr, nullptr, dv9.begin() + b);
}
```

Generally, the whole function should be rewritten, for now, it's not human-readable.

Reply:
This in on purpose. To have "?:" which casts third expression into type of second one.

Reply:
not applied, "?:" is on purpose